✨ demography: improve life expectancy metadata

lucasrodes opened the pull request:
Tracked here: https://github.com/owid/owid-issues/issues/1223

spoonerf commented on etl/steps/data/garden/demography/2023-09-27/survivor_percentiles.meta.yml:
```diff
@@ -19,9 +19,15 @@ tables:
           The age until which the << percentile >>% percentile of the population would survive until, if mortality rates remained constant, using data from a given interval.
         description_processing: |-
-          This was calculated with the method published in Alvarez and Vaupel (2023), with code provided by the authors. These estimates were regenerated for data from more recent years in the Human Mortality Database.
+          This was calculated with the method published in Alvarez and Vaupel (2023), with code provided by the authors:
```
Is it possible to link directly to the code? It doesn't look obvious where to find it from the paper and so it might be useful to help users find it.

spoonerf replied:
Ah I see they are at the top of the garden script, perhaps useful to note here too.

spoonerf reviewed (approved):
Looks good, just a minor comment!

lucasrodes commented:
I've updated the metadata fields based on https://github.com/owid/owid-issues/issues/1223#issuecomment-1753762658.

Note that I have split the proposed descriptions into `description_short` and `description_key`.
